File: api/internal/domain/quest.go
package domain

import "time"

type QuestGroup struct {
	ID     string `json:"id"`
	GameID string `json:"game_id"`
	Count  int    `json:"count"`
}

type QuestGroupCreate struct {
	GameID string `json:"game_id"`
	Count  int    `json:"count"`
}

type Quest struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`

	Money int `json:"money"`
	XP    int `json:"xp"`

	QuestType string `json:"quest_type"`
	GroupID   string `json:"group_id"`

	Lat float64 `json:"lat"`
	Lng float64 `json:"lng"`

	GameID string `json:"game_id"`

	CreatedAt time.Time `json:"created_at"`
}

type QuestCreate struct {
	Title       string `json:"title"`
	Description string `json:"description"`

	Money int `json:"money"`
	XP    int `json:"xp"`

	QuestType string `json:"quest_type"`
	GroupID   string `json:"group_id"`

	GameID string `json:"game_id"`

	Lat float64 `json:"lat"`
	Lng float64 `json:"lng"`
}

type ActiveQuest struct {
	ID        string    `json:"id"`
	QuestID   string    `json:"quest_id"`
	TeamID    string    `json:"team_id"`
	Complete  bool      `json:"complete"`
	CreatedAt time.Time `json:"created_at"`
}

type ActiveQuestFull struct {
	ID          string `json:"id"`
	QuestID     string `json:"quest_id"`
	Title       string `json:"title"`
	Description string `json:"description"`

	Money int `json:"money"`
	XP    int `json:"xp"`

	QuestType string `json:"quest_type"`
	GroupID   string `json:"group_id"`

	Lat float64 `json:"lat"`
	Lng float64 `json:"lng"`

	Complete bool `json:"complete"`

	GameID string `json:"game_id"`

	CreatedAt time.Time `json:"created_at"`
	StartedAt time.Time `json:"started_at"`
}

type ActiveQuestCreate struct {
	QuestID  string `json:"quest_id"`
	TeamID   string `json:"team_id"`
	Complete bool   `json:"complete"`
}

File: ios/Inertia/Inertia/Models/Quest.swift
//
//  Quest.swift
//  Inertia
//
//  Created by peony on 10/04/2024.
//

import Foundation

struct ActiveQuest: Codable, Identifiable {
    var id: String
    var questId: String
    var gameId: String
    var title: String
    var description: String
    var questType: String
    var money: Int
    var xp: Int
    var groupId: String
    var lat: Double?
    var lng: Double?
    var createdAt: String
    var complete: Bool
    
    enum CodingKeys: String, CodingKey {
        case id = "id"
        case questId = "quest_id"
        case gameId = "game_id"
        case title = "title"
        case description = "description"
        case questType = "quest_type"
        case money = "money"
        case xp = "xp"
        case groupId = "group_id"
        case lat = "lat"
        case lng = "lng"
        case createdAt = "created_at"
        case complete = "complete"
    }
    
    static func mock() -> Self {
        return ActiveQuest(id: "123", questId: "123", gameId: "123", title: "Wykasuj Sploya", description: "Wykasuj Sploya. Zrób to, zanim cię zje, tylko szybko! Nie chcesz aby cię dopadł Witecki. On jest głodny.", questType: "main", money: 0, xp: 300, groupId: "123", createdAt: "asdf", complete: false)
    }
}
